src/order/complete_lattice.lean

```diff
@@ -1000,6 +1000,18 @@ begin
   { simp_rw [infi_ge_eq_infi_nat_add, ←nat.add_assoc], },
 end
 
+lemma sup_supr_nat_succ (u : ℕ → α) : u 0 ⊔ (⨆ i, u (i + 1)) = ⨆ i, u i :=
```

Comment:
It would be nice to have a tactic that generates this kind of lemma for any inductive type, with `supr_and`, `supr_or`, `supr_sigma` etc as particular cases.